projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-util.ts

```diff
@@ -508,5 +508,28 @@ export class PivotUtil {
         }
     }
 
+    public static handleCountAggregator(columns: ColumnType[], value: IPivotValue, isSingleValue: boolean): void {
+        const isCountAggregator = value.aggregate.aggregator?.name?.toLowerCase() === 'count' || value.aggregate.aggregatorName?.toLowerCase() === 'count';
 
+        if ((value.dataType === GridColumnDataType.Currency || value.dataType === GridColumnDataType.Percent) && isCountAggregator) {
```

Comment:
These checks are a bit repetitive as is the code in the ```if``'s. The same ones are in `updateColumnDataTypeByAggregator` and in `generateColumnHierarchy`. Maybe just extract it in a common util method that gets the column type for a given PivotValue and reuse it. It would also make the code a bit more readable:

```
public static updateColumnTypeByAggregator(columns: ColumnType[], value: IPivotValue, isSingleValue: boolean): void {
  const targetColumnType = getColumnDataTypeForValue(value);   
   columns.forEach(column => {
                  if (column.field?.includes(value.member) || isSingleValue) {
                      column.dataType = targetColumnType;
                  }  
              });
}
```